hyperspy/drawing/utils.py

```diff
@@ -55,20 +59,30 @@ def contrast_stretching(data, saturated_pixels):
 
     Raises
     ------
-    ValueError if the value of `saturated_pixels` is out of the valid range.
+    ValueError if the value of `vmin` `vmax` is out of the valid range for
+    percentile calculation (in case of string values).
 
     """
-    # Sanity check
-    if saturated_pixels is None:
-        saturated_pixels = 0
-    if not 0 <= saturated_pixels <= 100:
-        raise ValueError(
-            "saturated_pixels must be a scalar in the range[0, 100]")
+    def _parse_value(value, value_name):
+        if value is None:
+            value = f'{getattr(preferences.Plot, value_name)}th'
+        if isinstance(value, str):
+            value = float(value.split("th")[0])
+        if not 0 <= value <= 100:
+            raise ValueError(f"{value_name} must be in the range[0, 100].")
+        return value
+
     if np.ma.is_masked(data):
         # If there is a mask, compressed the data to remove the masked data
         data = np.ma.masked_less_equal(data, 0).compressed()
-    vmin = np.nanpercentile(data, saturated_pixels / 2.)
-    vmax = np.nanpercentile(data, 100 - saturated_pixels / 2.)
+
+    # If vmin, vmax are float or int, we keep the value, if not we calculate
+    # the precentile value
+    if not isinstance(vmin, (float, int)):
+        vmin = np.nanpercentile(data, _parse_value(vmin, 'vmin'))
+    if not isinstance(vmax, (float, int)):
+        vmax = np.nanpercentile(data, 100 - _parse_value(vmax, 'vmax'))
```

Comment:
Wouldn't it be more intuitive to use `vmax="95th"` for the current equivalent of `vmax="5th"`. If yes, the widgets could use `IntRangeSlider` instead of `IntSlider`

Reply:
This is now done in https://github.com/hyperspy/hyperspy_gui_ipywidgets/pull/25.